### PR TITLE
Allow removing observers from components

### PR DESCRIPTION
**Pull request by jreij**

### CODE REVIEW

**Review by ozgur00 — DISMISSED** [review]

File: components-core/src/main/java/com/adyen/checkout/components/base/BaseActionComponent.java [hunk 1]
```diff
@@ -65,11 +65,21 @@ public void observe(@NonNull LifecycleOwner lifecycleOwner, @NonNull Observer<Ac
         mResultLiveData.observe(lifecycleOwner, observer);
     }
 
+    @Override
+    public void removeObservers(@NonNull LifecycleOwner lifecycleOwner) {
+        mResultLiveData.removeObservers(lifecycleOwner);
```
Comment (caiofaustino) [inline components-core/src/main/java/com/adyen/checkout/components/base/BaseActionComponent.java:70]: Why not use removeObserver() passing the observer reference instead? this way the merchant can have more granular control if they want.

Comment (jreij) [reply components-core/src/main/java/com/adyen/checkout/components/base/BaseActionComponent.java:70]: My thinking was that most of the times you have 1 observer per LifecycleOwner and you don't keep reference of your observers:

```kotlin
cardComponent.observe(this) { paymentComponentState ->
    // handle result
}
```

But I think we could add both and let the merchants choose what suits them better?

Comment (caiofaustino) [reply components-core/src/main/java/com/adyen/checkout/components/base/BaseActionComponent.java:70]: Sure

Comment (jreij) [reply components-core/src/main/java/com/adyen/checkout/components/base/BaseActionComponent.java:70]: fixed

**Review by caiofaustino — APPROVED** [review]

**Pull request merged**